zerver/openapi/zulip.yaml

```diff
@@ -545,6 +548,13 @@ paths:
               type: array
               items:
                 type: object
+                additionalProperties:
+                  type: string
+                  # This pattern was added as openapi-core
+                  # explicitly tries to convert everything to string
+                  # This prevents '{asd' to be treated as a allowed
+                  # string.
```

Comment:
I don't understand the issue here -- what would be different without this block?

Reply:
In the last commit, where we test bad narrow types, `openapi-core` converts array or object explicitly into string. So if narrow type is taken as `[0, 1]` then it will be treated as string(as it can be explicitly converted into string) and it will pass validation. So to prevent this I explicitly added regex pattern.

Reply:
What test specifically has this issue?